src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs

```diff
@@ -215,57 +215,28 @@ public void Serialize(ObjectWriter writer, CancellationToken cancellationToken)
                 Debug.Assert(ShouldSerialize(optionKey));
 
                 if (!_serializableOptions.Contains(optionKey.Option))
-                {
                     continue;
-                }
 
-                var kind = OptionValueKind.Null;
-                object? valueToWrite = null;
-                if (value != null)
+                OptionValueKind kind;
+                switch (value)
                 {
-                    switch (value)
-                    {
-                        case ICodeStyleOption codeStyleOption:
-                            if (optionKey.Option.Type.GenericTypeArguments.Length != 1)
-                            {
-                                continue;
-                            }
-
-                            kind = OptionValueKind.CodeStyleOption;
-                            valueToWrite = codeStyleOption;
-                            break;
-
-                        case NamingStylePreferences stylePreferences:
-                            kind = OptionValueKind.NamingStylePreferences;
-                            valueToWrite = stylePreferences;
-                            break;
-
-                        case string str:
-                            kind = OptionValueKind.String;
-                            valueToWrite = str;
-                            break;
-
-                        default:
-                            var type = value.GetType();
-                            if (type.IsEnum)
-                            {
-                                kind = OptionValueKind.Enum;
-                                valueToWrite = (int)value;
```

Comment:
this cast isn't necessary here due to teh line where we actually write the enum doing this already.